### PR TITLE
feature(reset): refactor the reset service to reset the cluster state completely

### LINKED ISSUE #218: refactor `reset` service to use etcd client
Currently, the reset service tries to delete resources through k8s API server.
https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/master/simulator/reset/reset.go

But, this way can only delete resources that are known by the simulator. We cannot delete resources that aren't supported by simulator or custom resources.

I'd suggest that use etcd client library and delete all the data created by our k8s API server.

In the storage config, we can define the prefix for the all keys created by the k8s API server.
https://pkg.go.dev/k8s.io/Kubernetes/pkg/storage/storagebackend
And etcd client library allows us to delete multiple data by giving prefix. 
https://pkg.go.dev/go.etcd.io/etcd/client/v3#WithPrefix

/kind cleanup
/triage accepted
/assign

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/area simulator
/kind feature
/kind cleanup

#### What this PR does / why we need it:

Currently, reset service tries to remove all **supported** resources. 
But, those who use other resources want to reset the cluster completely; including all unsupported resources.
(+ Especially, before running Scenario, we wanna clean up all resources to make sure no resources that affect scheduling in the cluster.)

---

And, besides those enhancements, we also can remove `DeleteCollection` method from all resource services.
So, it partly contributes to https://github.com/kubernetes-sigs/kube-scheduler-simulator/issues/139.

#### Which issue(s) this PR fixes:

Fixes #218

#### Special notes for your reviewer:

/label tide/merge-method-squash

### CODE REVIEW

**Comment by sanposhiho** [comment]
@196Ikuchil rebased. 🙏

**Review by 196Ikuchil — CHANGES_REQUESTED** [review]
Thank you for the PR.
I left some comments.

And this has a familiar problem.
It's a resource deletion of PriorityClass prefixed with "system-".

This is, after the resetting, `system-cluster-critical` and `system-node-critical` would also be deleted.
To avoid this, we should check the internal data of the etcd when deletion, or backup the data with these data, or restart with api-server, I think.
(This issue plagues us all time.. 🤔 )

File: simulator/reset/reset.go [hunk 1]
```diff
@@ -4,86 +4,65 @@ import (
...
-}
 
-// Reset cleans up all resources and scheduler configuration.
-func (s *Service) Reset(ctx context.Context) error {
-	eg, ctx := errgroup.WithContext(ctx)
-	nsList, err := s.client.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+	result, err := etcdClient.Get(context.Background(), "/"+k8sapiserver.EtcdPrefix, clientv3.WithPrefix())
 	if err != nil {
-		return xerrors.Errorf("list namespaces: %w", err)
+		return nil, xerrors.Errorf("delete all data in etcd: %w", err)
```
[the remark this answers is not in the document]
Comment (sanposhiho) [reply simulator/reset/reset.go]: Fixed. 🙏

**Review by 196Ikuchil — COMMENTED** [review]
Thank you!
This PR works fine on my local:+1:

/lgtm

**Pull request merged**